framework/src/main/java/org/apache/drill/test/framework/DrillQueryProfile.java

```diff
@@ -0,0 +1,241 @@
+package org.apache.drill.test.framework;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonTypeName("queryProfile")
+public class DrillQueryProfile {
```

Comment:
Not sure how easily maintainable would this be, and based on future protobuf changes, you'll need to maintain versions of this as well.

Reply:
```@JsonIgnoreProperties(ignoreUnknown = true)``` shields from failing to deserialize if there are protobuf changes (say, adding new fields or renaming existing fields). However, if the type of existing fields are changed, then changes are required in the framework. This is fair and helps catch the impact of protobuf changes.